heareval/predictions/task_predictions.py

```diff
@@ -1000,46 +1208,89 @@ def print_scores(grid_point_results: List[GridPointResult]):
     # Use the best model to compute test scores
     sort_grid_points(grid_point_results)
     best_grid_point = grid_point_results[0]
-    print()
     print(
         "Best validation score",
         best_grid_point.validation_score,
         best_grid_point.hparams,
         embedding_path,
     )
-    print(best_grid_point.model_path)
-
-    test_dataloader = dataloader_from_split_name(
-        "test",
-        embedding_path,
-        label_to_idx,
-        nlabels,
-        metadata["embedding_type"],
-        batch_size=conf["batch_size"],
-        in_memory=in_memory,
-    )
-    best_trainer = best_grid_point.trainer
-    # This hack is necessary because we use the best validation epoch to
-    # choose the event postprocessing
-    best_trainer.fit_loop.current_epoch = best_grid_point.epoch
-
-    test_results = best_trainer.test(
-        ckpt_path=best_grid_point.model_path, test_dataloaders=test_dataloader
-    )
-    assert len(test_results) == 1, "Should have only one test dataloader"
-    test_results = test_results[0]
 
-    test_results.update(
-        {
-            "validation_score": best_grid_point.validation_score,
-            "hparams": hparams_to_json(best_grid_point.hparams),
-            "postprocessing": best_grid_point.postprocessing,
-            "epoch": best_grid_point.epoch,
-            "time_in_min": best_grid_point.time_in_min,
-            "score_mode": best_grid_point.score_mode,
-            "embedding_path": str(embedding_path),
+    # Train the model using the best grid point configuration on the train+valid
```

Comment:
This long function is getting a bit confusing, maybe we break it down into several functions? I am not sure tho. I am wondering if there is a way to do more code re-use to simplify things

Reply:
Yes. Can we instead treat everything as a folded dataset? And datasets that have predefined train, test, validation are just a special case where there is a single configuration? I think we could get rid of some of these conditionals that way.